diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -185,7 +185,8 @@ fun ASTNode.isEol() = parent({ it.treeNext != null }, false)?.isFollowedByNewlin
  */
 fun ASTNode.isFollowedByNewline() =
     parent({ it.treeNext != null }, strict = false)?.let {
-        it.treeNext.elementType == WHITE_SPACE && it.treeNext.text.contains("\n")
+        it.isFollowedByNewlineCheck() ||
+                (it.treeNext.elementType == WHITE_SPACE && it.treeNext.treeNext.elementType == EOL_COMMENT && it.treeNext.treeNext.isFollowedByNewlineCheck())
```

Comment:
Please extract `it.treeNext` and `it.treeNext.treeNext` into separate variables for readability &mdash; or use `kotlin.run()`, e.g.:

```suggestion
                (probablyWhitespace.elementType == WHITE_SPACE && probablyWhitespace.treeNext.run { elementAfterWhitespace -> /* ... */ })
```

Reply:
Fixed